src/app/shared/services/dynamic-data/adapters/reactiveMemory.ts

```diff
@@ -128,8 +128,12 @@ export class ReactiveMemoryAdapater {
       const updatedDoc = await existingDoc.atomicPatch(data);
       return updatedDoc.toMutableJSON();
     } else {
-      await collection.insert(update);
+      await collection.insert(data);
       return update;
```

Comment:
nit(non-blocking)
nice catch on the insert, would probably be good to also return the updated doc as mutableJson (not that the return seems to be used anywhere)